ref-exchange/src/account_deposit.rs

```diff
@@ -46,7 +90,8 @@ impl Account {
     // [AUDIT_01]
     /// Returns amount of $NEAR necessary to cover storage used by this data structure.
     pub fn storage_usage(&self) -> Balance {
-        (MIN_ACCOUNT_DEPOSIT_LENGTH + self.tokens.len() as u128 * (MAX_ACCOUNT_BYTES + 16))
+        (INIT_ACCOUNT_STORAGE + 
+            self.tokens.len() as u64 * (ACC_ID_AS_KEY_STORAGE + U128_STORAGE)) as u128
             * env::storage_byte_cost()
     }
```

Comment:
So this should be returned from the `account.storage_used`.

When the contract plan to update the account or some other field owned by the account, it should do the following sequence:

1) Remember the initial storage: `let initial_storage = env::storage_used();`
2) Do modifications and save all data to the state.
3) Get the new storage used `let final_storage = env::storage_used();`
4) Update `account.storage_used` to the new value based on difference.
5) Verify the account has enough balance to store the new value.
6) Save the account again.

If everything is calculated correctly then each account will track its storage used across all pools and tokens.

Reply:
OK. In the second commit, I update the whole sotrage_used logic, and add corresponding sim test, see: sumulate_storage.rs.   

But there is a tricky thing, see account_deposit.rs#134, I don't know why these two numbers diff so much. Also, see test simulate_storage_max_account_id. there is the number from simulation.

Reply:
This seems like a pretty major change. How about we have one commit with versioned account separately.
And adding storage_used will be done separately.

I also don't think @evgenykuzyakov's proposal is great way because it's too easy to mess it up when more than one account is involved.

I suggest all the tracking and assets happen on update of the Account.

See https://github.com/ref-finance/ref-contracts/blob/ref-components/ref-components/src/account_impl.rs#L68

Reply:
OK, already revert the second commit. It is ready to merge now.